image/state.go

```diff
@@ -0,0 +1,65 @@
+package image
+
+import (
+	"sync"
+)
+
+var (
+	iSR = &imageStateMgr{
+		imgIDMap: make(map[ID]refCount),
+	}
```

Comment:
Rather than a global variable, the image state should be part of `Daemon`.

Maybe this functionality should be integrated into `image.Store` instead of keeping track of saves separately.